src/signals/incident-management/components/MapInput/index.js

```diff
@@ -10,52 +8,42 @@ import { formatMapLocation } from 'shared/services/map-location'
 import Label from 'components/Label'
 
 export const MapInput = (props) => {
-  const { name, display, onQueryResult } = props
-
-  // Can't use useCallback here, would break the rules of hooks
-  const render = ({ handler }) => {
-    const handlerValue = handler().value
-    const value = formatMapLocation(handlerValue)
-    const defaultCenter =
-      configuration.map.optionsBackOffice?.center || MAP_OPTIONS.center
-
-    const center = value?.coordinates || defaultCenter
-
-    const mapOptions = {
-      ...MAP_OPTIONS,
-      ...(configuration.map.optionsBackOffice || {}),
-      center,
-      zoom: 14,
-    }
-
-    const onLocationChange = (location) => {
-      onQueryResult(location)
-    }
-
-    return (
-      <div className="map-input">
-        <div className="mode_input map rij_verplicht">
-          <Label htmlFor={`form${name}`}>{display}</Label>
-
-          <div className="map-input__control invoer">
-            <MapInputComponent
-              id="map-input"
-              value={value}
-              onChange={onLocationChange}
-              mapOptions={mapOptions}
-              hasZoomControls
-            />
-          </div>
-        </div>
-      </div>
-    )
+  const { name, display, onQueryResult, value: valueFromProps } = props
+
+  const value = formatMapLocation(valueFromProps)
+  const defaultCenter =
+    configuration.map.optionsBackOffice?.center || MAP_OPTIONS.center
+
+  const center = value?.coordinates || defaultCenter
+
+  const mapOptions = {
+    ...MAP_OPTIONS,
+    ...(configuration.map.optionsBackOffice || {}),
+    center,
+    zoom: 14,
   }
 
-  render.propTypes = {
-    handler: PropTypes.func.isRequired,
+  const onLocationChange = (location) => {
+    onQueryResult(location)
   }
 
-  return render
+  return (
+    <div className="map-input">
+      <div className="mode_input map rij_verplicht">
+        <Label htmlFor={`form${name}`}>{display}</Label>
+
+        <div className="map-input__control invoer">
+          <MapInputComponent
+            id="map-input"
+            value={value}
+            onChange={onLocationChange}
+            mapOptions={mapOptions}
+            hasZoomControls
+          />
+        </div>
+      </div>
+    </div>
+  )
 }
 
```

Comment:
is het een idee om typescript van deze file te maken?

Reply:
gedaan hoor.Moest er wel een   // @ts-ignore voor gooien (of nog verder van de weg van pr fixes afraken).